src/components/evaluate.ts

```diff
@@ -0,0 +1,33 @@
+const expr = require('expression-eval');
+const cache = new Map();
+
+// Compile expression or return cached compiled expression
+function compile(str: string) {
+  const cached = cache.get(str);
+  if (cached) {
+    return cached;
+  } else {
+    const compiled = expr.compile(str);
+    cache.set(str, compiled);
+    return compiled;
+  }
+}
+
+export function evaluate(str: string, context: any, currentProp: string) {
+  // Transform `paths.*.*.responses.*.foo` into `foo`
+  const _context = {};
+  for (const prop in context) {
+    // Only consider properties ABOVE the current property in the schema
+    // (This enforces a top-to-bottom data dependency which is just nice.)
+    if (prop === currentProp) break;
+    const short = prop.split('.').pop() as string;
```

Comment:
How about using regular expression for this?

```
const [, short] =/\.([^.]+$)/.exec(prop);
```
didn't test it, but should be working fine

Reply:
So, here's my argument for using `split('.').pop()`

1. I think it's much more readable.
2. I doubt there is any noticeable performance benefit to the regex from the user's perspective.
3. I find people over-estimate their ability to use regular expressions 😛 , and thus regex's tend to be more error-prone than simple string functions. For instance, here you've overlooked a basic case: what if there is no periods in the word?

```js
let prop = "foo";
const short = prop.split('.').pop() // "foo"
const [, short2] =/\.([^.]+$)/.exec(prop); // Uncaught TypeError: /\.([^.]+$)/.exec is not a function or its return value is not iterable
```

You will find, many a time, I am the sworn enemy of the regular expression ☠️

Reply:
I find regu.ar expressions bring out the _worst_ of programmers...

https://github.com/developit/snarkdown/blob/master/src/index.js#L26

😆

Reply:
Yeah, obviously a TypeError is thrown, as exec returns null if no match is found.
If you do need to return a word regardless if dot is present or not... the regular expression would be even simpler :laughing: 
```
const [short] = /[^.]+$/.exec(prop) || [];
```

Reply:
I had such an approach as well, but then I spent a day or so and learn it.
I am so happy that ES2018 finally introduces lookbehind assertions and named capture groups.
Regular expressions FTW.

Anyway, I won't be able to convince you, so leave it as is.